content/docs/rfcs/66/README.md

```diff
@@ -0,0 +1,39 @@
+---
+slug: 66
+title: 66/WAKU2-METADATA
+name: Waku Metadata Protocol
+status: raw
+editor: Alvaro Revuelta <alrevuelta@status.im>
+contributors:
+---
+
+# Metadata Protocol
+
+Waku specifies a req/resp protocol that provides information about the node's medatadata. Such metadata is meant to be used
+by the node to decide if a peer is worth connecting or not. The node that makes the request, includes its metadata
+so that the receiver is aware of it, without requiring an extra interaction. The parameters are the following:
+* `networkId`: Unique identifier of the network that the node is running in.
+* `shards`: Shard indexes that the node is subscribed to.
+
+
+## Protocol id
+
+`/vac/waku/metadata/1.0.0`
+
+## Request
+
+```proto
+message WakuMetadataRequest {
+  uint32 network_id = 1;
+  repeated unt32 shards = 2;
+}
+```
+
+## Response
+
+```proto
+message WakuMetadataResponse {
+  uint32 network_id = 1;
```

Comment:
I would still make this `optional` in the proto3 definition (both in request and response).
This is not because populating the field is necessarily optional in the protocol, but because `proto3` limitations currently imply that there will be no way for us to know if the remote party populated the field or not without the `optional` keyword. As it stands, if this field is not explicitly populated it will always deserialise as network ID `0`, whether that was the intention or not.

Reply:
May be clear already, but just thought of a better way to state the implication. If we have a statement in an RFC such as:

> The `network_id` MUST be populated with the configured network ID...

there is no way for us to check this stipulation without the `optional` keyword. A bit silly in proto3 IMO.

Reply:
fixed 285670710300ce1e6a1a33053836c51e357942fc